newIDE/app/src/ObjectEditor/Editors/SpriteEditor/SpritesList.js

```diff
@@ -262,68 +272,85 @@ export default class SpritesList extends Component<Props, void> {
-      ) {
-        const originalSprite = direction.getSprite(resource.originalIndex);
-        copySpritePoints(originalSprite, sprite);
-        copySpritePolygons(originalSprite, sprite);
-      } else {
-        if (allDirectionSpritesHaveSamePoints) {
-          copySpritePoints(direction.getSprite(0), sprite);
-        }
-        if (allDirectionSpritesHaveSameCollisionMasks) {
-          copySpritePolygons(direction.getSprite(0), sprite);
+      const { resources, newMetadata, newName } = editResult;
+
+      const newDirection = new gd.Direction();
+      newDirection.setTimeBetweenFrames(direction.getTimeBetweenFrames());
+      newDirection.setLoop(direction.isLooping());
+      resources.forEach(resource => {
+        const sprite = new gd.Sprite();
+        sprite.setImageName(resource.name);
+        // Restore collision masks and points
+        if (
+          resource.originalIndex !== undefined &&
+          resource.originalIndex !== null
+        ) {
+          const originalSprite = direction.getSprite(resource.originalIndex);
+          copySpritePoints(originalSprite, sprite);
+          copySpritePolygons(originalSprite, sprite);
+        } else {
+          if (allDirectionSpritesHaveSamePoints) {
+            copySpritePoints(direction.getSprite(0), sprite);
+          }
+          if (allDirectionSpritesHaveSameCollisionMasks) {
+            copySpritePolygons(direction.getSprite(0), sprite);
+          }
         }
-      }
-      newDirection.addSprite(sprite);
-      sprite.delete();
-    });
+        newDirection.addSprite(sprite);
+        sprite.delete();
+      });
 
-    // Set metadata on the direction to allow editing again in the future.
-    if (newMetadata) {
-      newDirection.setMetadata(JSON.stringify(newMetadata));
-    }
+      // Set metadata on the direction to allow editing again in the future.
+      if (newMetadata) {
+        newDirection.setMetadata(JSON.stringify(newMetadata));
+      }
 
-    // Burst the ResourcesLoader cache to force images to be reloaded (and not cached by the browser).
-    resourcesLoader.burstUrlsCacheForResources(project, resourceNames);
-    onReplaceByDirection(newDirection);
+      // Burst the ResourcesLoader cache to force images to be reloaded (and not cached by the browser).
+      resourcesLoader.burstUrlsCacheForResources(project, resourceNames);
+      onReplaceByDirection(newDirection);
 
-    // If a name was specified in the external editor, use it for the animation.
-    if (newName) {
-      onChangeName(newName);
+      // If a name was specified in the external editor, use it for the animation.
+      if (newName) {
+        onChangeName(newName);
+      }
+      newDirection.delete();
+    } catch (error) {
+      this.setState({ externalEditorOpened: false });
+      console.error(
+        'An exception was thrown when launching or reading resources from the external editor:',
+        error
+      );
+      showErrorBox({

```

Comment:
maybe use showAlert() even if it shouldn't happen much ?

Reply:
I would love but unfortunately the parents are class components :/